## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 10 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.cynthion</li>
    <li>python312Packages.cynthion.dist</li>
    <li>python312Packages.luna-soc</li>
    <li>python312Packages.luna-soc.dist</li>
    <li>python312Packages.luna-usb</li>
    <li>python312Packages.luna-usb.dist</li>
    <li>python313Packages.luna-soc</li>
    <li>python313Packages.luna-soc.dist</li>
    <li>python313Packages.luna-usb</li>
    <li>python313Packages.luna-usb.dist</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:x: 19 packages failed to build:</summary>
  <ul>
    <li>pony-corral</li>
    <li>ponyc</li>
    <li>python312Packages.deal</li>
    <li>python312Packages.deal-solver</li>
    <li>python312Packages.deal-solver.dist</li>
    <li>python312Packages.deal.dist</li>
    <li>python312Packages.icontract</li>
    <li>python312Packages.icontract.dist</li>
    <li>python312Packages.pylddwrap</li>
    <li>python312Packages.pylddwrap.dist</li>
    <li>python313Packages.deal</li>
    <li>python313Packages.deal-solver</li>
    <li>python313Packages.deal-solver.dist</li>
    <li>python313Packages.deal.dist</li>
    <li>python313Packages.icontract</li>
    <li>python313Packages.icontract.dist</li>
    <li>python313Packages.pylddwrap</li>
    <li>python313Packages.pylddwrap.dist</li>
    <li>ugarit</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 78 packages built:</summary>
  <ul>
    <li>acl2</li>
    <li>alive2</li>
    <li>circt</li>
    <li>circt.dev</li>
    <li>circt.lib</li>
    <li>cynthion</li>
    <li>cynthion.dist</li>
    <li>dafny</li>
    <li>foundry</li>
    <li>ghidra-extensions.kaiju</li>
    <li>glasgow</li>
    <li>glasgow.dist</li>
    <li>hal-hardware-analyzer</li>
    <li>hexxy</li>
    <li>iprover</li>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>kdePackages.kinfocenter</li>
    <li>kdePackages.kinfocenter.debug</li>
    <li>kdePackages.kinfocenter.dev</li>
    <li>kdePackages.kinfocenter.devtools</li>
    <li>klee</li>
    <li>libbap</li>
    <li>ocamlPackages.z3</li>
    <li>oils-for-unix</li>
    <li>python312Packages.amaranth</li>
    <li>python312Packages.amaranth-boards</li>
    <li>python312Packages.amaranth-boards.dist</li>
    <li>python312Packages.amaranth-soc</li>
    <li>python312Packages.amaranth-soc.dist</li>
    <li>python312Packages.amaranth.dist</li>
    <li>python312Packages.angr</li>
    <li>python312Packages.angr.dist</li>
    <li>python312Packages.angrcli</li>
    <li>python312Packages.angrcli.dist</li>
    <li>python312Packages.angrop</li>
    <li>python312Packages.angrop.dist</li>
    <li>python312Packages.bap</li>
    <li>python312Packages.bap.dist</li>
    <li>python312Packages.claripy</li>
    <li>python312Packages.claripy.dist</li>
    <li>python312Packages.miasm</li>
    <li>python312Packages.miasm.dist</li>
    <li>python312Packages.model-checker</li>
    <li>python312Packages.model-checker.dist</li>
    <li>python312Packages.z3-solver (python312Packages.z3-solver.dev ,python312Packages.z3-solver.lib ,python312Packages.z3-solver.python)</li>
    <li>python313Packages.amaranth</li>
    <li>python313Packages.amaranth-boards</li>
    <li>python313Packages.amaranth-boards.dist</li>
    <li>python313Packages.amaranth-soc</li>
    <li>python313Packages.amaranth-soc.dist</li>
    <li>python313Packages.amaranth.dist</li>
    <li>python313Packages.bap</li>
    <li>python313Packages.bap.dist</li>
    <li>python313Packages.claripy</li>
    <li>python313Packages.claripy.dist</li>
    <li>python313Packages.model-checker</li>
    <li>python313Packages.model-checker.dist</li>
    <li>python313Packages.z3-solver (python313Packages.z3-solver.dev ,python313Packages.z3-solver.lib ,python313Packages.z3-solver.python)</li>
    <li>sail-riscv</li>
    <li>sby</li>
    <li>simbaplusplus</li>
    <li>solc</li>
    <li>supabase-cli</li>
    <li>symcc</li>
    <li>tlaps</li>
    <li>vampire</li>
    <li>vdrPlugins.markad</li>
    <li>vdrPlugins.streamdev</li>
    <li>z3</li>
    <li>z3-tptp</li>
    <li>z3.dev</li>
    <li>z3.lib</li>
    <li>z3.python</li>
    <li>z3_4_13</li>
    <li>z3_4_13.dev</li>
    <li>z3_4_13.lib</li>
    <li>z3_4_13.python</li>
  </ul>
</details>

----

ponyc looks unrelated:

```
[ 76%] Building C object src/ponyc/CMakeFiles/ponyc.dir/main.c.o
warning: Skipping impure flag -march=native because NIX_ENFORCE_NO_NATIVE is set
cp: cannot stat 'libstdc++.a': No such file or directory
```

deal-solver appears to be a bigger problem:

```
_________________ test_expr_asserts_ok[len({4, 5, 5, 6}) >= 3] _________________
[gw20] linux -- Python 3.12.9 /nix/store/wz0j2zi02rvnjiz37nn28h3gfdq61svz-python3-3.12.9/bin/python3.12

check = 'len({4, 5, 5, 6}) >= 3'

    @pytest.mark.parametrize('check', [
        # compare
        '{1, 2, 3} == {1, 2, 3}',
        '{1, 2, 3} != {1, 2, 3, 4}',
        '{1, 2, 3} == {3, 1, 2}',
        '{1, 2, 3} == {3, 2, 1, 1, 2}',
        'set() != {1}',
        '{1} != set()',
        'set() == set()',
    
        # compare mismatching types
        '{1} != [1]',
        '{1} != {1: 2}',
        '{1} != "hi"',
        '{1} != 1',
        '{1} != True',
    
        # 'len({7, 9, 9, 9, 11}) == 3',
    
        # operations
        '10 in {3, 6, 10, 17}',
        '2 not in {3, 6, 10, 17}',
        '{1, 2} | {2, 3} == {1, 2, 3}',
        '{1, 2} ^ {2, 3} == {1, 3}',
        '{1, 2} & {2, 3} == {2}',
    
        # methods
        '{1, 2}.union({2, 3}) == {1, 2, 3}',
        '{1, 2}.intersection({2, 3}) == {2}',
        '{1, 2}.symmetric_difference({2, 3}) == {1, 3}',
        '{1, 2}.difference({2, 3}) == {1}',
        '{1, 2}.copy() == {1, 2}',
    
        # is* methods
        '{1, 2}.issubset({1, 2, 3})',
        'not {1, 2, 3}.issubset({1, 2})',
        '{1, 2, 3}.issuperset({1, 2})',
        'not {1, 2}.issuperset({1, 2, 3})',
        '{1, 2}.isdisjoint({3, 4})',
        'not {1, 2}.isdisjoint({2, 3})',
    
        # functions
        'len({True}) >= 1',
        'len({4}) >= 1',
        'len({4, 5, 5, 6}) >= 3',
        'len(set()) == 0',
        'set({1, 2}) == {1, 2}',
    ])
    def test_expr_asserts_ok(check: str) -> None:
        assert eval(check)
        text = """
            from typing import List
            def f():
                assert {}
        """
        text = text.format(check)
        theorem = prove_f(text)
>       assert theorem.conclusion is Conclusion.OK
E       AssertionError: assert <Conclusion.SKIP: 'skipped'> is <Conclusion.OK: 'proved!'>
E        +  where <Conclusion.SKIP: 'skipped'> = Proof(conclusion=<Conclusion.SKIP: 'skipped'>, description='assertion', error=ProveError('timeout'), example=None).conclusion
E        +  and   <Conclusion.OK: 'proved!'> = Conclusion.OK

tests/test_types/test_set.py:67: AssertionError
```

--- a/pkgs/applications/science/logic/z3/default.nix
+++ b/pkgs/applications/science/logic/z3/default.nix
@@ -153,6 +153,10 @@ let
   fixupPatches = dir: map (patch: replaceVars patch { inherit dir; });
 in
 {
+  z3_4_14 = common {
+    version = "4.14.1";
+    sha256 = "sha256-pTsDzf6Frk4mYAgF81wlR5Kb1x56joFggO5Fa3G2s70=";
+  };
   z3_4_13 = common {
     version = "4.13.4";
     sha256 = "sha256-8hWXCr6IuNVKkOegEmWooo5jkdmln9nU7wI8T882BSE=";

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -17065,6 +17065,7 @@ with pkgs;
   };
 
   inherit (callPackages ../applications/science/logic/z3 { python = python3; })
+    z3_4_14
     z3_4_13
     z3_4_12
     z3_4_11

